[WIP] cljc migration

iku000888 opened the pull request:
- [x] move cljx files to cljc + use reader conditionals
- [ ] replace cljx plugin with cljsee
- [ ] test it out

Implements https://github.com/plumatic/schema/issues/219

iku000888 commented:
Status: Can confirm the clj tests passing. cljs tests have some issue

iku000888 commented:
Had to upgrade cljsbuild to compile the tests

<details>
<summary> output</summary>

$ lein with-profile dev cljsbuild test
Retrieving lein-cljsbuild/lein-cljsbuild/1.1.7/lein-cljsbuild-1.1.7.pom from clojars
Retrieving lein-cljsbuild/lein-cljsbuild/1.1.7/lein-cljsbuild-1.1.7.jar from clojars
Compiling ClojureScript...
Retrieving cljsbuild/cljsbuild/1.1.7/cljsbuild-1.1.7.pom from clojars
Retrieving cljsbuild/cljsbuild/1.1.7/cljsbuild-1.1.7.jar from clojars
Compiling ["target/main.js"] from ["src/clj" "target/generated/src/cljs"]...
WARNING: No such namespace: clojure.lang.Compiler, could not locate clojure/lang/Compiler.cljs, clojure/lang/Compiler.cljc, or JavaScript source providing "clojure.lang.Compiler" at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: Use of undeclared Var clojure.lang.Compiler/CHAR_MAP at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/parse-arity-spec at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/emit-defrecord at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/set-compile-fn-validation! at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/normalized-defn-args at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/primitive-sym? at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/maybe-split-first at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-schema-form at line 1404 target/generated/src/cljs/schema/core.cljs
Successfully compiled ["target/main.js"] in 11.443 seconds.
Compiling ["target/unit-test.js"] from ["src/clj" "test/clj" "target/generated/src/cljs" "target/generated/test/cljs"]...
WARNING: No such namespace: clojure.lang.Compiler, could not locate clojure/lang/Compiler.cljs, clojure/lang/Compiler.cljc, or JavaScript source providing "clojure.lang.Compiler" at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: Use of undeclared Var clojure.lang.Compiler/CHAR_MAP at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/parse-arity-spec at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/emit-defrecord at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/set-compile-fn-validation! at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/normalized-defn-args at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/primitive-sym? at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/maybe-split-first at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-schema-form at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: schema.core/either is deprecated. at line 243 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 250 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 254 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 263 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 297 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 532 target/generated/test/cljs/schema/core_test.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 666 target/generated/test/cljs/schema/core_test.cljs
WARNING: Use of undeclared Var macros/normalized-metadata at line 666 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 773 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 847 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 866 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 1040 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1235 is being replaced at line 1237 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1237 is being replaced at line 1240 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1240 is being replaced at line 1243 target/generated/test/cljs/schema/core_test.cljs
Successfully compiled ["target/unit-test.js"] in 10.605 seconds.
Compiling ["target/unit-test-no-assert.js"] from ["src/clj" "test/clj" "target/generated/src/cljs" "target/generated/test/cljs"]...
WARNING: No such namespace: clojure.lang.Compiler, could not locate clojure/lang/Compiler.cljs, clojure/lang/Compiler.cljc, or JavaScript source providing "clojure.lang.Compiler" at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: Use of undeclared Var clojure.lang.Compiler/CHAR_MAP at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/parse-arity-spec at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/emit-defrecord at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/set-compile-fn-validation! at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/normalized-defn-args at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/primitive-sym? at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/maybe-split-first at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-schema-form at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: schema.core/either is deprecated. at line 243 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 250 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 254 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 263 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 297 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 532 target/generated/test/cljs/schema/core_test.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 666 target/generated/test/cljs/schema/core_test.cljs
WARNING: Use of undeclared Var macros/normalized-metadata at line 666 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 773 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 847 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 866 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 1040 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1235 is being replaced at line 1237 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1237 is being replaced at line 1240 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1240 is being replaced at line 1243 target/generated/test/cljs/schema/core_test.cljs
Successfully compiled ["target/unit-test-no-assert.js"] in 9.076 seconds.
Running ClojureScript test: unit

Testing schema.core-test 

FAIL in (schema.core-test/constrained-test) (:)
expected: (= (quote (constrained Int odd?)) (s/explain s))
  actual: (not (= (constrained Int odd?) (constrained Int cljs$core$odd?)))

FAIL in (schema.core-test/fn-name-test) (:)
expected: (= "odd?" (utils/fn-name odd?))
  actual: (not (= "odd?" "cljs$core$odd?"))

FAIL in (schema.core-test/fn-name-test) (:)
expected: (= "foo-bar" (utils/fn-name foo-bar))
  actual: (not (= "foo-bar" "schema$core-test$foo-bar"))

FAIL in (schema.core-test/fn-name-test) (:)
expected: (= "foo" (utils/fn-name (fn foo [x] (+ x x))))
  actual: (not (= "foo" "schema$core-test$fn-name-test-test-$-foo"))

FAIL in (schema.core-test/conditional-test) (:)
as simple constraint
expected: (= (quote (conditional odd? Int)) (s/explain (s/conditional odd? s/Int)))
  actual: (not (= (conditional odd? Int) (conditional cljs$core$odd? Int)))

FAIL in (schema.core-test/conditional-test) (:)
as simple constraint
expected: (= (quote (conditional odd? Int weird?)) (s/explain (s/conditional odd? s/Int (quote weird?))))
  actual: (not (= (conditional odd? Int weird?) (conditional cljs$core$odd? Int weird?)))

FAIL in (schema.core-test/issue-310-error-wrap-cache) (:)
expected: (= "([:first (not (integer? :a))] [:second (not (integer? nil))])" (pr-str (s/check (->CacheTest s/Int) (->ItemTest :a nil))))
  actual: (not (= "([:first (not (integer? :a))] [:second (not (integer? nil))])" "([:first (not (cljs$core$integer? :a))] [:second (not (cljs$core$integer? nil))])"))

FAIL in (schema.core-test/issue-310-error-wrap-cache) (:)
expected: (= "([:first (not (integer? :a))] nil)" (pr-str (s/check (->CacheTest [s/Int]) (->ItemTest :a nil))))
  actual: (not (= "([:first (not (integer? :a))] nil)" "([:first (not (cljs$core$integer? :a))] nil)"))

FAIL in (schema.core-test/issue-310-error-wrap-cache) (:)
expected: (= "([:first (not (integer? :a))] [:second [(not (integer? nil))]])" (pr-str (s/check (->CacheTest [s/Int]) (->ItemTest :a [nil]))))
  actual: (not (= "([:first (not (integer? :a))] [:second [(not (integer? nil))]])" "([:first (not (cljs$core$integer? :a))] [:second [(not (cljs$core$integer? nil))]])"))

Testing schema.experimental.abstract-map-test 

Testing schema.coerce-test 

Ran 98 tests containing 445 assertions.
Testing complete: 9 failures, 0 errors.

Ran 98 tests containing 445 assertions.
Testing complete: 9 failures, 0 errors.
Running ClojureScript test: unit-no-assert

Testing schema.core-test 

FAIL in (schema.core-test/constrained-test) (:)
expected: (= (quote (constrained Int odd?)) (s/explain s))
  actual: (not (= (constrained Int odd?) (constrained Int cljs$core$odd?)))

FAIL in (schema.core-test/fn-name-test) (:)
expected: (= "odd?" (utils/fn-name odd?))
  actual: (not (= "odd?" "cljs$core$odd?"))

FAIL in (schema.core-test/fn-name-test) (:)
expected: (= "foo-bar" (utils/fn-name foo-bar))
  actual: (not (= "foo-bar" "schema$core-test$foo-bar"))

FAIL in (schema.core-test/fn-name-test) (:)
expected: (= "foo" (utils/fn-name (fn foo [x] (+ x x))))
  actual: (not (= "foo" "schema$core-test$fn-name-test-test-$-foo"))

FAIL in (schema.core-test/conditional-test) (:)
as simple constraint
expected: (= (quote (conditional odd? Int)) (s/explain (s/conditional odd? s/Int)))
  actual: (not (= (conditional odd? Int) (conditional cljs$core$odd? Int)))

FAIL in (schema.core-test/conditional-test) (:)
as simple constraint
expected: (= (quote (conditional odd? Int weird?)) (s/explain (s/conditional odd? s/Int (quote weird?))))
  actual: (not (= (conditional odd? Int weird?) (conditional cljs$core$odd? Int weird?)))

FAIL in (schema.core-test/issue-310-error-wrap-cache) (:)
expected: (= "([:first (not (integer? :a))] [:second (not (integer? nil))])" (pr-str (s/check (->CacheTest s/Int) (->ItemTest :a nil))))
  actual: (not (= "([:first (not (integer? :a))] [:second (not (integer? nil))])" "([:first (not (cljs$core$integer? :a))] [:second (not (cljs$core$integer? nil))])"))

FAIL in (schema.core-test/issue-310-error-wrap-cache) (:)
expected: (= "([:first (not (integer? :a))] nil)" (pr-str (s/check (->CacheTest [s/Int]) (->ItemTest :a nil))))
  actual: (not (= "([:first (not (integer? :a))] nil)" "([:first (not (cljs$core$integer? :a))] nil)"))

FAIL in (schema.core-test/issue-310-error-wrap-cache) (:)
expected: (= "([:first (not (integer? :a))] [:second [(not (integer? nil))]])" (pr-str (s/check (->CacheTest [s/Int]) (->ItemTest :a [nil]))))
  actual: (not (= "([:first (not (integer? :a))] [:second [(not (integer? nil))]])" "([:first (not (cljs$core$integer? :a))] [:second [(not (cljs$core$integer? nil))]])"))

Testing schema.experimental.abstract-map-test 

Testing schema.coerce-test 

Ran 98 tests containing 445 assertions.
Testing complete: 9 failures, 0 errors.

Ran 98 tests containing 445 assertions.
Testing complete: 9 failures, 0 errors.
Error encountered performing task 'cljsbuild' with profile(s): 'dev'
</details>

iku000888 commented:
Status:

- `lein test` passes without any error
- Many warnings during cljs build that were not present before the change. Need to investigate.

iku000888 commented:
<details>
<summary>warnings</summary>

WARNING: No such namespace: clojure.lang.Compiler, could not locate clojure/lang/Compiler.cljs, clojure/lang/Compiler.cljc, or JavaScript source providing "clojure.lang.Compiler" at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: Use of undeclared Var clojure.lang.Compiler/CHAR_MAP at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/parse-arity-spec at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/emit-defrecord at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/set-compile-fn-validation! at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/normalized-defn-args at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/primitive-sym? at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/maybe-split-first at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-schema-form at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: clojure.lang.Compiler, could not locate clojure/lang/Compiler.cljs, clojure/lang/Compiler.cljc, or JavaScript source providing "clojure.lang.Compiler" at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: Use of undeclared Var clojure.lang.Compiler/CHAR_MAP at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/parse-arity-spec at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/emit-defrecord at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/set-compile-fn-validation! at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/normalized-defn-args at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/primitive-sym? at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/maybe-split-first at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-schema-form at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: schema.core/either is deprecated. at line 243 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 250 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 254 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 263 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 297 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 533 target/generated/test/cljs/schema/core_test.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 667 target/generated/test/cljs/schema/core_test.cljs
WARNING: Use of undeclared Var macros/normalized-metadata at line 667 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 774 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 848 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 867 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 1041 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1236 is being replaced at line 1238 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1238 is being replaced at line 1241 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1241 is being replaced at line 1244 target/generated/test/cljs/schema/core_test.cljs
WARNING: No such namespace: clojure.lang.Compiler, could not locate clojure/lang/Compiler.cljs, clojure/lang/Compiler.cljc, or JavaScript source providing "clojure.lang.Compiler" at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: Use of undeclared Var clojure.lang.Compiler/CHAR_MAP at line 49 target/generated/src/cljs/schema/utils.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/parse-arity-spec at line 1088 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/emit-defrecord at line 1166 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/set-compile-fn-validation! at line 1179 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1269 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1270 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/normalized-defn-args at line 1332 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/process-fn- at line 1334 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/primitive-sym? at line 1337 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-arrow-schematized-element at line 1398 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/maybe-split-first at line 1400 target/generated/src/cljs/schema/core.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: Use of undeclared Var macros/extract-schema-form at line 1404 target/generated/src/cljs/schema/core.cljs
WARNING: schema.core/either is deprecated. at line 243 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 250 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 254 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 263 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 297 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 533 target/generated/test/cljs/schema/core_test.cljs
WARNING: No such namespace: macros, could not locate macros.cljs, macros.cljc, or JavaScript source providing "macros" at line 667 target/generated/test/cljs/schema/core_test.cljs
WARNING: Use of undeclared Var macros/normalized-metadata at line 667 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/either is deprecated. at line 774 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 848 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 867 target/generated/test/cljs/schema/core_test.cljs
WARNING: schema.core/both is deprecated. at line 1041 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1236 is being replaced at line 1238 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1238 is being replaced at line 1241 target/generated/test/cljs/schema/core_test.cljs
WARNING: v at line 1241 is being replaced at line 1244 target/generated/test/cljs/schema/core_test.cljs

</details>

w01fe commented:
Thanks for tackling this!  Just a heads up that i'm traveling and won't be able to look at this for the next two weeks or so.

iku000888 commented:
Understood. Safe travels!

iku000888 commented:
Status:

- Fixed cljs tests
- Dropped cljsee for debugging

Plan:
- Try adding back cljsee for backwards compat
- Investigate minimum ClojureScript version supported.
(As to not defeats the point of using cljsee)

```
$ lein with-profile dev cljsbuild test
Compiling ClojureScript...
Compiling ["target/unit-test-no-assert.js"] from ["src/clj" "test/clj" "src/cljx" "test/cljx"]...
WARNING: schema.core/either is deprecated. at line 243 test/cljx/schema/core_test.cljc
WARNING: schema.core/either is deprecated. at line 250 test/cljx/schema/core_test.cljc
WARNING: schema.core/both is deprecated. at line 254 test/cljx/schema/core_test.cljc
WARNING: schema.core/both is deprecated. at line 263 test/cljx/schema/core_test.cljc
WARNING: schema.core/both is deprecated. at line 297 test/cljx/schema/core_test.cljc
WARNING: schema.core/either is deprecated. at line 533 test/cljx/schema/core_test.cljc
WARNING: schema.core/either is deprecated. at line 774 test/cljx/schema/core_test.cljc
WARNING: schema.core/both is deprecated. at line 848 test/cljx/schema/core_test.cljc
WARNING: schema.core/both is deprecated. at line 867 test/cljx/schema/core_test.cljc
WARNING: schema.core/both is deprecated. at line 1041 test/cljx/schema/core_test.cljc
WARNING: v at line 1236 is being replaced at line 1238 test/cljx/schema/core_test.cljc
WARNING: v at line 1238 is being replaced at line 1241 test/cljx/schema/core_test.cljc
WARNING: v at line 1241 is being replaced at line 1244 test/cljx/schema/core_test.cljc
Successfully compiled ["target/unit-test-no-assert.js"] in 20.903 seconds.
Running ClojureScript test: unit

Testing schema.core-test 

Testing schema.experimental.abstract-map-test 

Testing schema.coerce-test 

Ran 98 tests containing 445 assertions.
Testing complete: 0 failures, 0 errors.

Ran 98 tests containing 445 assertions.
Testing complete: 0 failures, 0 errors.
Running ClojureScript test: unit-no-assert

Testing schema.core-test 

Testing schema.experimental.abstract-map-test 

Testing schema.coerce-test 

Ran 98 tests containing 445 assertions.
Testing complete: 0 failures, 0 errors.

Ran 98 tests containing 445 assertions.
Testing complete: 0 failures, 0 errors.
```

w01fe commented on project.clj:
```diff
@@ -4,33 +4,37 @@
-  :profiles {:dev {:dependencies [[org.clojure/clojure "1.7.0"]
-                                  [org.clojure/clojurescript "0.0-2760"]
-                                  [org.clojure/tools.nrepl "0.2.5"]
+  :profiles {:dev {:dependencies [[org.clojure/clojure "1.8.0"]
```
Does this mean it no longer works on 1.7? (there was already a profile for 1.8).

w01fe replied:
never mind, I missed your last comment

w01fe commented on src/cljx/schema/macros.cljc:
```diff
@@ -1,6 +1,7 @@
 (ns schema.macros
   "Macros and macro helpers used in schema.core."
   (:require
+   [cljs.analyzer.api]
```
Is this actually used?

w01fe commented on src/cljx/schema/macros.cljc:
```diff
@@ -201,7 +205,8 @@
   (when-let [bad-meta (seq (filter (or (meta bind) {}) [:tag :s? :s :schema]))]
-    (throw (RuntimeException. (str "Meta not supported on bindings, put on fn name" (vec bad-meta)))))
+    (throw #?(:clj (RuntimeException. (str "Meta not supported on bindings, put on fn name" (vec bad-meta)))
+              :cljs (js/Error. (str "Meta not supported on bindings, put on fn name" (vec bad-meta))))))
```
This file is all Clojure-side, so I don't think it needs to be cljc.

iku000888 commented:
@w01fe Thanks for taking a look!

Would get back on this in a week or two!

frenchy64 commented:
Closing as we eventually got this done. Thanks!

iku000888 commented:
Great news! Should https://github.com/plumatic/schema/issues/219 be closed?
